port/unix/omrsysinfo.c

```diff
@@ -2359,74 +2359,86 @@ omrsysinfo_set_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, ui
 
 	Trc_PRT_sysinfo_set_limit_Entered(resourceID, limit);
 
-	if (OMRPORT_RESOURCE_ADDRESS_SPACE == resourceRequested) {
+	switch (resourceRequested) {
+	case OMRPORT_RESOURCE_FILE_DESCRIPTORS:
+		resource = RLIMIT_NOFILE;
+		break;
+	case OMRPORT_RESOURCE_ADDRESS_SPACE:
 #if !defined(OMRZTPF)
 		resource = RLIMIT_AS;
 #else /* !defined(OMRZTPF) */
 		/* z/TPF does not have virtual address support */
 		rc = -1;
-		Trc_PRT_sysinfo_set_limit_Exit(rc);
-		return rc;
 #endif /* !defined(OMRZTPF) */
-	} else if (OMRPORT_RESOURCE_CORE_FILE == resourceRequested) {
+		break;
+	case OMRPORT_RESOURCE_CORE_FILE:
 #if !defined(OMRZTPF)
 		resource = RLIMIT_CORE;
 #else /* !defined(OMRZTPF) */
 		rc = -1;
-		Trc_PRT_sysinfo_set_limit_Exit(rc);
-		return rc;
 #endif /* !defined(OMRZTPF) */
+		break;
+	default:
+		break;
 	}
 
-	switch (resourceRequested) {
-	case OMRPORT_RESOURCE_ADDRESS_SPACE:
-		/* FALLTHROUGH */
-	case OMRPORT_RESOURCE_CORE_FILE: {
-#if !defined(OMRZTPF)
-		rc = getrlimit(resource, &lim);
-		if (-1 == rc) {
-			portLibrary->error_set_last_error(portLibrary, errno, findError(errno));
-			Trc_PRT_sysinfo_getrlimit_error(resource, findError(errno));
+	if (0 == rc) {
+		switch (resourceRequested) {
+		case OMRPORT_RESOURCE_FILE_DESCRIPTORS:
+#if defined(OMRZTPF)
+			rc = -1;
 			break;
-		}
+#endif /* defined(OMRZTPF) */
```

Comment:
Since this is falling through, the ztpf case is already handled in the fallthrough code to return -1

Reply:
Okay.